test/dns-test-srv/main.go

```diff
@@ -6,15 +6,60 @@
 package main
 
 import (
+	"encoding/json"
 	"fmt"
+	"io/ioutil"
 	"net"
+	"net/http"
 	"os"
+	"strings"
+	"sync"
 	"time"
 
 	"github.com/letsencrypt/boulder/Godeps/_workspace/src/github.com/miekg/dns"
 )
 
-func dnsHandler(w dns.ResponseWriter, r *dns.Msg) {
+type testSrv struct {
+	mu         *sync.RWMutex
+	txtRecords map[string]string
+}
+
+type setRequest struct {
+	Host  string `json:"host"`
+	Value string `json:"value"`
+}
+
+func (ts *testSrv) setTXT(w http.ResponseWriter, r *http.Request) {
+	if r.URL.Path != "/set-txt" {
+		http.NotFound(w, r)
+		return
+	} else if r.Method != "POST" {
+		w.WriteHeader(405)
+		return
+	}
+	msg, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
```

Comment:
`http.Server` will automatically close request bodies meaning we don't really have to.

> // For server requests the Request Body is always non-nil
    // but will return EOF immediately when no body is present.
    // The Server will close the request body. The ServeHTTP
    // Handler does not need to.